homeassistant/components/netatmo/climate.py

```diff
@@ -82,90 +148,264 @@ def temperature_unit(self):
     @property
     def current_temperature(self):
         """Return the current temperature."""
-        return self._data.current_temperature
+        return self._data.room_status[self._room_id]['current_temperature']
 
     @property
     def target_temperature(self):
         """Return the temperature we try to reach."""
-        return self._target_temperature
+        return self._data.room_status[self._room_id]['target_temperature']
 
     @property
     def current_operation(self):
         """Return the current state of the thermostat."""
-        state = self._data.thermostatdata.relay_cmd
-        if state == 0:
+        state = self._data.room_status[self._room_id]['heating_status']
+        if state is False:
             return STATE_IDLE
-        if state == 100:
+        if state is True:
             return STATE_HEAT
+        return STATE_UNKNOWN
+
+    @property
+    def operation_list(self):
+        """Return the operation modes list."""
+        return self._operation_list
+
+    @property
+    def operation_mode(self):
+        """Return current operation ie. heat, cool, idle."""
+        return self._operation_mode
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        module_type = self._data.room_status[self._room_id]['module_type']
+        if module_type not in ('NATherm1', 'NRV'):
+            return {}
+        state_attributes = {
+            "home_id": self._data.homedata.gethomeId(self._data.home),
+            "room_id": self._room_id,
+            "setpoint_default_duration": self._data.setpoint_duration,
+            "away_temperature": self._data.away_temperature,
+            "hg_temperature": self._data.hg_temperature,
+            "operation_mode": self._operation_mode,
+            "module_type": module_type,
+            "module_id": self._data.room_status[self._room_id]['module_id']
+        }
+        if module_type == 'NATherm1':
```

Comment:
Can you move NATherm1/NRV to constants since they are used a lot?

Reply:
Moved in 7142f52.